registrations/models.py

```diff
@@ -236,6 +239,59 @@ def is_user_editable_resources(self):
         )
 
 
+class RegistrationUser(models.Model):
+    email = models.EmailField(verbose_name=_("E-mail"))
+
+    registration = models.ForeignKey(
+        Registration, on_delete=models.CASCADE, related_name="registration_users"
+    )
+
+    def can_be_edited_by(self, user):
+        """Check if current registration user can be edited by the given user"""
+        if user.is_superuser:
+            return True
+        return user.is_admin_of(self.registration.event.publisher)
+
+    def send_invitation(self):
+        locale = "fi"
```

Comment:
Would it be possible to send the UI locale when doing the API request when creating or updating a RegistrationUser, store that too into RegistrationUser and use it here to override the invitation email language (as long as it would be one of the supported langues in which case a fallback like this current one could be used)?

Reply:
I just want to confirm before commiting that I get this rights. Is the idea to add `language` field to RegistrationUser model and then use it as email language? e.g.

```
class RegistrationUser(models.Model):
    email = models.EmailField(verbose_name=_("E-mail"))

    registration = models.ForeignKey(
        Registration, on_delete=models.CASCADE, related_name="registration_users"
    )

    language = models.ForeignKey(
        Language,
        on_delete=models.SET_NULL,
        null=True,
        blank=True,
        related_name="registration_user_language",
    )

    def _get_language_pk(self):
        if self.language:
            return self.language.pk
        return "fi"

    def send_invitation(self):
        locale = self._get_language_pk()

        with translation.override(locale):
            event_name = self.registration.event.name
            event_type_id = self.registration.event.type_id

            email_variables = {
                "email": self.email,
                "event": event_name,
                "event_type_id": event_type_id,
                "linked_events_ui_locale": locale,
                "linked_events_ui_url": settings.LINKED_EVENTS_UI_URL,
                "linked_registrations_ui_locale": locale,
                "linked_registrations_ui_url": settings.LINKED_REGISTRATIONS_UI_URL,
                "registration_id": self.registration.id,
            }

            invitation_template = "signup_list_rights_granted.html"
            invitation_subject = _(
                "Rights granted to the participant list - %(event_name)s"
            ) % {"event_name": event_name}

            rendered_body = render_to_string(invitation_template, email_variables)

            try:
                send_mail(
                    invitation_subject,
                    rendered_body,
                    get_email_noreply_address(),
                    [self.email],
                    html_message=rendered_body,
                )
            except SMTPException:
                pass
```

In that case it would be nice to allow only Languages where `service_languauge` is set to True. Similar logic is used in email sending to signups
```
class RegistrationUserSerializer(serializers.ModelSerializer):
    language = serializers.PrimaryKeyRelatedField(
        queryset=Language.objects.filter(service_language=True),
        many=False,
        required=False,
    )

    class Meta:
        model = RegistrationUser
        fields = ["id", "email"]
```

Reply:
Yes, this is what I had in mind in terms of the logic (didn't think too much into the details of how it would be best implemented, but your suggestions looks good)